### PR TITLE
BX-4239 Order Status Feed

### PR DESCRIPTION
**Pull request by kdonthi**
- Updated proto file
- Created sample code going through the lifecycle of a transaction

### CODE REVIEW

File: example/transaction/main.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    print("starting settle funds")
+    post_settle_response = await p.post_settle(owner_address=public_key, market=marketAddr, base_token_wallet=baseTokenWallet, quote_token_wallet=quoteTokenWallet, open_orders_address=open_orders)
+    print("settle transaction created successfully")
+
+    signed_settle_tx = signing.sign_tx(post_settle_response.transaction)
+
+    post_submit_response = await p.post_submit(transaction=signed_settle_tx, skip_pre_flight=True)
+    print("response signature for settle received: " + post_submit_response.signature)
+
+async def just_seeing():
```
Comment (kdonthi) [inline example/transaction/main.py]: Remove this

File: example/transaction/main.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+async def ws():
+    async with provider.ws() as api:
+        async with provider.ws() as api2:
+            await order_lifecycle(api, api2)
+
+async def grpc():
+    async with provider.grpc() as api:
+        await order_lifecycle(api, api)
+
+async def order_lifecycle(p1: provider.Provider, p2: provider.Provider):
```
Comment (kdonthi) [inline example/transaction/main.py:40]: add comments

File: example/transaction/main.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    oss = p2.get_order_status_stream(market=marketAddr, owner_address=public_key)
+    task1 = asyncio.create_task(oss.__anext__())
```
Comment (kdonthi) [inline example/transaction/main.py]: Try w/o task?

File: example/transaction/main.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
+import asyncio
+import os
+import random
+
+import async_timeout
+
+from bxserum import proto, provider
+from bxserum.transaction import signing
+
+public_key=os.getenv("PUBLIC_KEY")
```
Comment (aspin) [inline example/transaction/main.py:10]: formatting: download https://github.com/psf/black for formatting these example files

**Review by aspin — APPROVED** [review]
merging for feature enhancement

**Pull request merged**